odoo/modules/registry.py

```diff
@@ -973,7 +974,15 @@ def cursor(self, /, readonly=False):
 
         connection = self._db
         if readonly and self._db_readonly is not None:
+            connection_rw = connection
             connection = self._db_readonly
+            try:
+                return connection.cursor()
+            except psycopg2.OperationalError:
+                _logger.warning('Failed to open a readonly cursor, falling back to read-write cursor')
+                # Deactivate the readonly feature until restart / worker recycling.
+                self._db_readonly = None
+                return connection_rw.cursor()
```

Comment:
I understand the motives for minimal diff, but since this is critical code I would rather we have the simplest code possible
```py
if readonly and self._db_readonly is not None:
    try:
        return self._db_readonly.cursor()
    except psycopg2.OperationalError:
        self._db_readonly = None
        _logger.warning(...)
return self._db.cursor()
```